augur/platform_connector.py

```diff
@@ -0,0 +1,663 @@
+#SPDX-License-Identifier: MIT
```

Comment:
As I noted in our discussion, I think this is a *really* wise refactoring. This puts all the API logic and a lot of the *shared* worker database logic *that is related to the API connection* and puts it in one place. 

If CLI logic, or other CORE Augur functions require API access in the future, (i.e., if we wanted to make PyPi packages out of all the workers, having this logic here would make that a **lot** easier.(

You know from our meeting last week that I like this change/refactor for those reasons.